### PR TITLE
SCX RunAsProvider ExecuteShellCommand

### PR DESCRIPTION
**Pull request by Cyb3rWard0g**
This hunting query uses Auditd security events collected via the Syslog data connector to explore the use of the [SCX RunAsProvider Invoke_ExecuteShellCommand](https://github.com/microsoft/SCXcore/blob/master/source/code/providers/support/runasprovider.cpp#L137) to execute any UNIX/Linux command using the /bin/sh shell.

SCXcore, started as the Microsoft Operations Manager UNIX/Linux Agent, is now used in a host of products including Microsoft Operations Manager. Microsoft Azure, and Microsoft Operations Management Suite.

SCX has a support provider named [RunAsProvider](https://github.com/microsoft/SCXcore/blob/master/source/code/providers/support/runasprovider.cpp). This provider has a few classes:
* [ExecuteCommand](https://github.com/microsoft/SCXcore#runas-provider-executecommand)
* [ExecuteShellCommand](https://github.com/microsoft/SCXcore#runas-provider-executeshellcommand)
* [ExecuteScript](https://github.com/microsoft/SCXcore#runas-provider-executescript)

Based on [OMIGOD: Critical Vulnerabilities in OMI Affecting Countless Azure Customers](https://www.wiz.io/blog/omigod-critical-vulnerabilities-in-omi-azure) by [Wiz](https://twitter.com/wiz_io) , `ExecuteShellCommand` was used in the HTTP request to test  `CVE-2021-38647`. 

```
<s:Body>
      <p:ExecuteShellCommand_INPUT xmlns:p="http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/SCX_OperatingSystem">
         <p:command>id</p:command>
         <p:timeout>0</p:timeout>
      </p:ExecuteShellCommand_INPUT>
   </s:Body>
```  

This was derived from initial testing while executing commands via `/opt/omi/bin/omicli` and exploring responses.

```
/opt/omi/bin/omicli --hostname 192.168.1.1 -u azureuser -p Password1 iv root/scx { SCX_OperatingSystem } ExecuteShellCommand { command 'id' timeout 0 }
```

Using the same template provided in the blog post by Wiz, we prepared a quick test:

```
<s:Body>
      <p:ExecuteShellCommand_INPUT xmlns:p="http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/SCX_OperatingSystem">
         <p:command>echo 'Hola MSTIC'</p:command>
         <p:timeout>0</p:timeout>
      </p:ExecuteShellCommand_INPUT>
   </s:Body>
```  

We set the SCX logging to  `verbose`  

```
/opt/microsoft/scx/bin/tools/scxadmin -log-set all verbose
```

and we were able to capture the activity on the OMI server side in the `scx.log`:

```
tail -f /var/opt/microsoft/scx/log/scx.log
```

image

image

Next, we checked our `Sysmon for Linux` and `auditd` logs in our lab environment and identified where the commands were being executed from:

 
image

image

We then put together the following query to validate our testing. The query is part of this PR.

image

# References:
* https://www.wiz.io/blog/omigod-critical-vulnerabilities-in-omi-azure
* https://docs.microsoft.com/en-us/system-center/scom/manage-security-administer-crossplat-agent?view=sc-om-2019
* https://github.com/microsoft/SCXcore
* https://github.com/microsoft/SCXcore/blob/master/source/code/providers/support/runasprovider.cpp#L137

### CODE REVIEW

File: Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  - InitialAccess
+  - Execution
+relevantTechniques:
+  - T1190
+  - T1203
+tags:
+  - SimuLand
+query: |
+  Syslog
+  | parse SyslogMessage with "type=" EventType " audit(" * "): " EventData
```
Comment (shainw) [inline Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml:26]: Right after the parse, I would recommend for performance you filter with has on eventdata patterns you match on later, it improves the query time a bunch on larger data sets - | where EventData has "AUOMS_EXECVE" and EventData has '/var/opt/microsoft/scx/tmp'

File: Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  | project TimeGenerated, EventType, Computer, EventData 
+  // Extract AUOMS_EXECVE details from EventData
+  | where EventType =~ "AUOMS_EXECVE"
+  | parse EventData with * "syscall=" syscall " syscall_r=" * " success=" success " exit=" exit " a0" * " ppid=" ppid " pid=" pid " audit_user=" audit_user " auid=" auid " user=" user " uid=" uid " group=" group " gid=" gid "effective_user=" effective_user " euid=" euid " set_user=" set_user " suid=" suid " filesystem_user=" filesystem_user " fsuid=" fsuid " effective_group=" effective_group " egid=" egid " set_group=" set_group " sgid=" sgid " filesystem_group=" filesystem_group " fsgid=" fsgid " tty=" tty " ses=" ses " comm=\"" comm "\" exe=\"" exe "\"" * "cwd=\"" cwd "\"" * "name=\"" name "\"" * "cmdline=\"" cmdline "\" containerid=" containerid
```
Comment (shainw) [inline Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml:30]: I see you have multiple user fields (user, audit_user,effected user...).  Does it make sense to map them all to the Account entity at the end?  Do these end up being the same account in this case?  If not, you can have multiple columnames for the same field mapping.  entityType: Account, identifier: FullName (or whatever proper identifier if not FullName) and columnName: User and then another set with the same identifer but columnName: audit_user

**Review by shainw — CHANGES_REQUESTED** [review]
1 recommended change and 1 potential change depending on what is in the user fields, otherwise good.

File: Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml (hunk 1 above)
Comment (Cyb3rWard0g) [reply Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml:26]: I like that.

File: Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml (hunk 2 above)
Comment (Cyb3rWard0g) [reply Hunting Queries/Syslog/SCXRunAsProviderExecuteShellCommand.yml:29]: done. I used the `user` field for that. It will be root always.

**Review by shainw — APPROVED** [review]

**Pull request merged**